src/main/java/internet/shop/dao/impl/DriverDaoImpl.java

```diff
@@ -1,45 +1,120 @@
     }
 
     @Override
     public List<Driver> getAll() {
-        return Storage.getDrivers();
+        List<Driver> driverList = new ArrayList<>();
+        String query = "SELECT * FROM drivers where deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                driverList.add(createDriver(resultSet));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from drivers DB", exception);
+        }
+        return driverList;
     }
 
     @Override
     public Driver update(Driver driver) {
-        List<Driver> dbDriver = Storage.getDrivers();
-        IntStream.range(0, dbDriver.size())
-                .filter(i -> dbDriver.get(i).equals(driver))
-                .findFirst()
-                .ifPresent(i -> dbDriver.set(i, driver));
-        return driver;
+        String query = "UPDATE drivers SET"
+                + " name = ?, license_number = ?"
+                + " where deleted = false and id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.setLong(3, driver.getId());
+
+            if (preparedStatement.executeUpdate() > 0) {
+                return driver;
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + driver + " into drivers", exception);
+        }
+        throw new RuntimeException("No driver " + driver + " to update in drivers DB");
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getDrivers().removeIf(i -> i.getId().equals(id));
+        String query = "UPDATE drivers SET deleted = true "
+                + "where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: "
+                    + id + " into drivers", exception);
+        }
+    }
+
+    private Driver createDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject(1, Long.class));
+        driver.setName(resultSet.getObject(2, String.class));
+        driver.setLicenceNumber(resultSet.getObject(3, String.class));
```

Comment:
use column names instead of `1`, `2`, `3`

Reply:
done